bot/snakegame.py

```diff
@@ -0,0 +1,149 @@
+from random import randint
+
+
+class SnakeGame:
+    """
+    Simple Snake game
+    """
+
+    def __init__(self, board_dimensions):
+        self.board_dimensions = board_dimensions
+        self.restart()
+
+    def restart(self):
+        """
+        Restores game to default state
+        """
+
+        self.snake = Snake(positions=[[2, 2], [2, 1]])
+        self.putFood()
+        self.score = 0
+
+    def __str__(self):
+        """
+        Draw the board
+        """
+        # create empty board
+        # board_string = "+" + "-" * (self.board_dimensions[1]) + "+" + "\n"
+        board_string = ""
+        for i in range(self.board_dimensions[0]):
+            # row_string = "|"
+            row_string = ""
+            # draw snake
+            for j in range(self.board_dimensions[1]):
+                if [i, j] == self.snake.positions[0]:
+                    row_string += ":python:"  # head
+                elif [i, j] in self.snake.positions:
+                    row_string += "🐍"
+                elif [i, j] == self.food:
+                    row_string += "🍕"
+                else:
+                    row_string += "◻️"
+            # row_string += "|\n"
+            board_string += row_string + "\n"
+        # board_string += "+" + "-" * (self.board_dimensions[1]) + "+\n"
+
+        return board_string
+
+    def move(self, direction):
+        """
+        Executes one movement.
+        Returns information about the movement:
+        "ok", "forbidden", "lost", "food".
+        """
+
+        direction_dict = {
+            "right": (0, 1),
+            "left": (0, -1),
+            "up": (-1, 0),
+            "down": (1, 0)
+        }
+        move_status = self.snake.move(direction_dict[direction])
+
+        if self.isLost():
+            move_status = "lost"
+            self.restart()
+
+        if self.isEating():
+            self.snake.grow()
+            move_status = "grow"
+            self.putFood()
+            self.score += 1
+
+        return move_status
+
+    def isLost(self):
```

Comment:
This should use python property decorator! So it would look like 
```py
@property
def lost(self):
```
Applies to other `is` functions too!